tools/lib/provision.py

```diff
@@ -262,7 +264,24 @@ def main(options):
         else:
             print("No need to regenerate the test DB.")
 
-        run(["./manage.py", "compilemessages"])
+        # Check if translation files have been modified or not
+        sha1sum = hashlib.sha1()
+        paths = glob.glob('static/locale/*/LC_MESSAGES/*.po')
+
+        for path in paths:
+            sha1sum.update(subprocess_text_output(['cat', path]))
+        sha1sum.update(subprocess_text_output(['cat', 'zerver/management/commands/compilemessages.py']))
+
+        new_compilemessages_hash = sha1sum.hexdigest()
+        run(['touch', 'var/last_compilemessages_hash'])
```

Comment:
It seems cleaner to catch the "does not exist" exception than touch the file first.